x-pack/test/observability_onboarding_api_integration/tests/elastic_agent/config.spec.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { load } from 'js-yaml';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+  const observabilityOnboardingApiClient = getService('observabilityOnboardingApiClient');
+
+  async function callApi({ onboardingId }: { onboardingId: string }) {
+    return await observabilityOnboardingApiClient.logMonitoringUser({
+      endpoint: 'GET /internal/observability_onboarding/elastic_agent/config',
+      params: {
+        query: {
+          onboardingId,
+        },
+      },
+    });
+  }
```

Comment:
should we also be passing in an API key here and make assertions in the rendered yaml in the test below?

Reply:
I haven't been successful with that part. The supertest is built with an url like `http://user:changeme@localhost:5601` and when trying to set headers with the ApiKey they get overriden by that config in the url of supertest.
Do you have any suggestions?

Reply:
the only other way i can think of is to use another API/library to make the REST call: `fetch`, `axios`, `http.client` etc.

Reply:
Will play around with those and check if we can get that covered as well 😃